draft-ietf-quic-tls.md

```diff
@@ -1107,6 +1107,21 @@ of the handshake, they cannot be fully trusted until the handshake completes,
 and reliance on them should be minimized.  However, any tampering with the
 parameters will cause the handshake to fail.
 
+Endpoints MUST NOT send this extension in a TLS connection that does not use
+QUIC (such as the use of TLS with TCP defined in {{!TLS13}}).  A fatal
+unsupported_extension alert MUST be sent if this extension is received when the
+transport is not QUIC.
+
+
+## Removing the EndOfEarlyData Message {#remove-eoed}
+
+Negotiating the quic_transport_parameters extension also modifies TLS to remove
+the EndOfEarlyData message.  QUIC does not rely on this message to mark the end
```

Comment:
My preference goes to simply stating that we do not use EOED in QUIC, rather than stating that the use of a particular extension signals the omission (I assume that is what @siyengar suggested).

My read is that the proposed text expects the stacks to be implemented in either of the following two ways, which seems like an unnecessary complication to me:
* making TLS stacks depend on an extension defined outside of TLS
* let the server-side QUIC stack modify the flag of the TLS stack (signaling the omission) when it processes ClientHello

Reply:
I share those thoughts - while it is unlikely that a random TCP connection would send a QUIC extension, QUIC cannot mandate what other protocols should and should not do. Furthermore, a QUIC client might not know for sure that what the other end supports, and even if it thinks that it does, the might not remain true after a random intern pushed the wrong button.